how-to: update hardware-and-sofware-requirements (#2658)
pingcap/docs: hardware-and-software-requirements.md

> ## About
> 

Reviewer: ```suggestion
```

> - A large number of TiDB tests have been run on the CentOS 7.3 system, and in our community there are a lot of best practices in which TiDB is deployed on the Linux operating system. Therefore, it is recommended to deploy TiDB on CentOS 7.3 or later.
> - The support for the Linux operating systems above includes the deployment and operation in physical servers as well as in major virtualized environments like VMware, KVM and XEN.

## Software recommendations

### Control Machine

| Software | Version |
| :--- | :--- |
| sshpass | 1.06 or later |
| TiUP | 0.6.2 or later |

> **Note:**
>

> > It is required that you [deploy TiUP](/tiup/tiup-cluster.md) on the Control Machine to operate and manage TiDB clusters.

Reviewer: ```suggestion
> It is required that you [deploy TiUP](/production-deployment-using-tiup.md#第-2-步在中控机上安装-tiup-组件) on the Control Machine to operate and manage TiDB clusters.
```
Waiting on https://github.com/pingcap/docs-cn/pull/3228/files#r433695667

Reviewer: Please see https://github.com/pingcap/docs-cn/pull/3228/files#r433772498 to fix the docs. @ran-huang

> > It is required that you [deploy TiUP on the Control Machine](/tiup/tiup-documentation-guide.md) to operate and manage TiDB clusters.

Author: ```suggestion
> It is required that you deploy TiUP on the Control Machine to operate and manage TiDB clusters.
```

> - A large number of TiDB tests have been run on the CentOS 7.3 system, and in our community there are a lot of best practices in which TiDB is deployed on the Linux operating system. Therefore, it is recommended to deploy TiDB on CentOS 7.3 or later.
> - The support for the Linux operating systems above includes the deployment and operation in physical servers as well as in major virtualized environments like VMware, KVM and XEN.

## Software recommendations

### Control Machine

| Software | Version |
| :--- | :--- |
| sshpass | 1.06 or later |
| TiUP | 0.6.2 or later |

> **Note:**
>

> > [Deploy TiUP](/tiup/tiup-cluster.md) on the Control Machine to operate and manage TiDB clusters.

Reviewer: ```suggestion
> It is required that you [deploy TiUP](/tiup/tiup-cluster.md) on the Control Machine to operate and manage TiDB clusters.
```

### Target machines

| Software | Version |
| :--- | :--- |
| sshpass | 1.06 or later |
| numa | 2.0.12 or later |

## Server recommendations

> You can deploy and run TiDB on the 64-bit generic hardware server platform in the Intel x86-64 architecture or the hardware server platform in the ARM architecture. The requirements and recommendations about server hardware configuration (ignoring the resources occupied by the operating system itself) for development, test, and production environments are as follows:

Reviewer: ```suggestion
You can deploy and run TiDB on the 64-bit generic hardware server platform in the Intel x86-64 architecture or on the hardware server platform in the ARM architecture. The requirements and recommendations about server hardware configuration (ignoring the resources occupied by the operating system itself) for development, test, and production environments are as follows:
```

> - In the test environment, the TiDB and PD instances can be deployed on the same server.
> - For performance-related test, do not use low-performance storage and network hardware configuration, in order to guarantee the correctness of the test result.
> - For the TiKV server, it is recommended to use NVMe SSDs to ensure faster reads and writes.

> > - If you only want to verify the features, follow [Quick Start Guide for TiDB](/quick-start-with-tidb.md) to deploy TiDB and test the features on a single machine.

Reviewer: ```suggestion
> - If you only want to test and verify the features, follow [Quick Start Guide for TiDB](/quick-start-with-tidb.md) to deploy TiDB on a single machine.
```
